packages/integrations/preact/src/index.ts

```diff
@@ -1,94 +1,68 @@
 import type { AstroIntegration, AstroRenderer, ViteUserConfig } from 'astro';
+import preact, {type PreactPluginOptions as VitePreactPluginOptions} from '@preact/preset-vite';
+import { fileURLToPath } from 'node:url';
+
+const babelCwd = new URL('../', import.meta.url);
 
 function getRenderer(development: boolean): AstroRenderer {
 	return {
 		name: '@astrojs/preact',
 		clientEntrypoint: development ? '@astrojs/preact/client-dev.js' : '@astrojs/preact/client.js',
 		serverEntrypoint: '@astrojs/preact/server.js',
-		jsxImportSource: 'preact',
-		jsxTransformOptions: async () => {
-			// @ts-expect-error types not found
-			const plugin = await import('@babel/plugin-transform-react-jsx');
-			const jsx = plugin.default?.default ?? plugin.default;
-			return {
-				plugins: [jsx({}, { runtime: 'automatic', importSource: 'preact' })],
-			};
-		},
 	};
 }
 
-function getCompatRenderer(development: boolean): AstroRenderer {
+export type Options =Pick<VitePreactPluginOptions, 'include' | 'exclude'> & { compat?: boolean };
+// TODO: Add back compat support -- how would this work in the new world?
```

Comment:
I think the plugin now has compat support? Or are they missing features with this refactor?

Reply:
Nope it has it, will remove the comment.